### PR TITLE
Add spinner component (#268)

### PR DESCRIPTION
**Pull request by SomayeEbrahimi**

### CODE REVIEW

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Spinner/BitSpinner.razor.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    break;
+                case SpinnerSize.Small:
+                    classSize = $"{RootElementClass}-small-{VisualClassRegistrar()}";
+                    break;
+                case SpinnerSize.Large:
+                    classSize = $"{RootElementClass}-large-{VisualClassRegistrar()}";
+                    break;
+            }
+
+            return classSize;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Spinner/BitSpinner.razor.cs:36]: You can simplify the method like this:
```cs
string classSize = string.Empty;

switch (Size)
{
    case SpinnerSize.XSmall:
        classSize = "xSmall";
        break;
    case SpinnerSize.Small:
        classSize = "small";
        break;
    case SpinnerSize.Medium:
        classSize = "medium";
        break;
    case SpinnerSize.Large:
        classSize = "large";
        break;
}

return $"{RootElementClass}-{classSize}-{VisualClassRegistrar()}";
```

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Spinner/BitSpinner.razor.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        private string GetClassLabelPosition()
+        {
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Spinner/BitSpinner.razor.cs:40]: You can simplify this too

File: src/Client/Web/Bit.Client.Web.BlazorUI/Styles/Fluent/Spinner/bit-spinner.fluent.scss [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    border-width: 1.5px;
+    border-style: solid;
+    border-color: $Primary $BlizzardBlue $BlizzardBlue;
+    border-image: initial;
+    animation-name: bit-spinner;
+    animation-duration: 1.3s;
+    animation-iteration-count: infinite;
+}
+
+.bit-spinner-large-fluent div:first-child {
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Styles/Fluent/Spinner/bit-spinner.fluent.scss:50]: please use SCSS features in these situations.
you could ask khashaayaar.

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor [hunk 4]
```diff
@@ -1,4 +1,26 @@
 ﻿@page "/component"
+<style>
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor:2]: please move these styles into the designated SCSS files.

**Review by msynk — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Spinner/BitSpinner.razor [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿@namespace Bit.Client.Web.BlazorUI
+@inherits BitComponentBase
+
+<div @ref="RootElement"
+     style="@StyleBuilder.Value"
+     class="@ClassBuilder.Value">
+    <div></div>
```
[the remark this answers is not in the document]
Comment (SomayeEbrahimi) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Spinner/BitSpinner.razor:7]: No, it used for spinner

File: src/Client/Web/Bit.Client.Web.BlazorUI/Styles/Fluent/Spinner/bit-spinner.fluent.scss [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    align-items: flex-end;
+}
+
+.bit-spn-right-fluent {
+    flex-direction: row-reverse;
+    align-items: flex-end;
+}
+
+.bit-spn {
+    & > div {
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Styles/Fluent/Spinner/bit-spinner.fluent.scss:39]: these can be moved inside the first declaration of ".bit-spn".

File: src/Client/Web/Bit.Client.Web.BlazorUI/Styles/Fluent/Spinner/bit-spinner.fluent.scss [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        &:first-child {
+            width: 12px;
+            height: 12px;
+        }
+    }
+}
+
+.bit-spn {
+    & > div {
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Styles/Fluent/Spinner/bit-spinner.fluent.scss:91]: these can be moved inside the first declaration of ".bit-spn".

**Review by msynk — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Spinner/BitSpinner.razor [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    @if (!string.IsNullOrEmpty(Label))
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Spinner/BitSpinner.razor:8]: change this line into this:
```cs
@if (string.IsNullOrEmpty(Label) is false)
```

**Review by msynk — CHANGES_REQUESTED** [review]

**Review by msynk — APPROVED** [review]

**Pull request merged**